src/prpl.ts

```diff
@@ -12,15 +12,175 @@
  * http://polymer.github.io/PATENTS.txt
  */
 
+import * as fs from 'fs';
 import * as http from 'http';
+import * as path from 'path';
 import * as send from 'send';
 import * as url from 'url';
 
-export function handler(root?: string): (
+import * as capabilities from './capabilities';
+import * as push from './push';
+
+// The subset of the polymer.json specification that we care about for serving.
+// https://www.polymer-project.org/2.0/docs/tools/polymer-json
+// https://github.com/Polymer/polymer-project-config/blob/master/src/index.ts
+export interface ProjectConfig {
+  entrypoint?: string;
+  builds?: {
+    name?: string,
+    browserCapabilities?: capabilities.BrowserCapability[],
+  }[];
+}
+
+// Matches "/foo/bar.png" but not "/foo.png/bar".
+const hasFileExtension = new RegExp('\\.[^/]*$');
+
+// TODO Service worker location should be configurable.
+const isServiceWorker = new RegExp('service-worker.js$');
```

Comment:
If not constructing a regex dynamically, just use regex literals: `/service-worker.js$/`